src/centralstore/incremental/NodeManager.cpp

```diff
@@ -0,0 +1,179 @@
+#include "NodeManager.h"
+
+#include <sys/stat.h>
+
+#include "NodeBlock.h"  // To setup node DB
+#include "PropertyLink.h"
+
+NodeManager::NodeManager(std::string mode) {
+    std::ios_base::openmode openMode = std::ios::trunc;  // default is Trunc mode which overrides the entire file
+    if (mode == "app") {
+        openMode = std::ios::app;  // if app, open in append mode
+        this->nodeIndex = readNodeIndex();
+    }
+    NodeBlock::nodesDB =
+        new std::fstream(NodeManager::NODE_DB_PATH, std::ios::in | std::ios::out | openMode | std::ios::binary);
+    PropertyLink::propertiesDB =
+        new std::fstream(PropertyLink::DB_PATH, std::ios::in | std::ios::out | openMode | std::ios::binary);
+    // TODO: set PropertyLink nextPropertyIndex after validating by modulus check from file number of bytes
```

Comment:
When we have a `TODO` comment lets annotate it with the github id as well so that we can know who has made the `TODO` comment. For example,

``// TODO (tmkasun): set PropertyLink nextPropertyIndex after validating by modulus check from file number of bytes``

Reply:
Thanks,
Fixed in https://github.com/miyurud/jasminegraph/pull/104/commits/9f35b95f13fccc96e0b3963e753b5efde0c6b89f

Reply:
Done